backend/pkg/database/gorm_common.go

```diff
@@ -192,6 +192,110 @@ func (gr *GormRepository) GetUsers(ctx context.Context) ([]models.User, error) {
 	return sanitizedUsers, result.Error
 }
 
+func (gr *GormRepository) GetUser(ctx context.Context, userID uuid.UUID) (*models.User, error) {
+	var dbUser models.User
+	var user models.User
+	result := gr.GormClient.WithContext(ctx).First(&dbUser, userID)
+	if result.Error != nil {
+		return nil, result.Error
+	}
+
+	user.ID = dbUser.ID
+	user.FullName = dbUser.FullName
+	user.Email = dbUser.Email
+	user.Username = dbUser.Username
+	user.Role = dbUser.Role
+
+	// Populate ACLs for the user
+	var acls []models.UserPermission
+	if err := gr.GormClient.WithContext(ctx).
+		Where("user_id = ?", user.ID).
+		Find(&acls).Error; err != nil {
+		return nil, err
+	}
+	user.Permissions = make(map[string]map[string]bool)
+
+	for _, acl := range acls {
+		if _, exists := user.Permissions[acl.TargetUserID.String()]; !exists {
+			user.Permissions[acl.TargetUserID.String()] = make(map[string]bool)
+		}
+		user.Permissions[acl.TargetUserID.String()][string(acl.Permission)] = true
+	}
```

Comment:
Ideally this would be done with GORM Associations -- so we could use a single query. 

https://gorm.io/docs/create.html#Create-With-Associations
https://gorm.io/docs/query.html#Joins-Preloading